src/backup_restore.cpp

```diff
@@ -157,6 +157,33 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
         return;
     }
 
+    const std::string& l_srcFruPath =
+        jsonUtility::getFruPathFromJson(m_sysCfgJsonObj, i_srcPath);
+    const std::string& l_dstFruPath =
+        jsonUtility::getFruPathFromJson(m_sysCfgJsonObj, i_dstPath);
+    if (l_srcFruPath.empty() || l_dstFruPath.empty())
+    {
+        logging::logMessage(
+            "Couldn't find either source or destination FRU path.");
+        return;
+    }
+
+    const std::string& l_srcServiceName =
+        m_sysCfgJsonObj["frus"][l_srcFruPath][0]["serviceName"];
+    const std::string& l_dstServiceName =
```

Comment:
Have a json utility function to get service name?

Reply:
added new utility API to get service name.